### DISCUSSION TITLE
vcpkg ignores CMAKE_CXX_COMPILER_LAUNCHER

### DISCUSSION THREAD

**Discussion by wrobelda** [microsoft/vcpkg#21595] [Ideas]
**Describe the bug**
I added `-DCMAKE_CXX_COMPILER_LAUNCHER=ccache` to my cmake switches, hoping that vcpkg would pick it up to build own packages.

However, it seems that vcpkg ignores it completely. This is suboptimal, because the alternatives come down to spoofing the C/CXX compiler (https://ccache.dev/manual/4.4.2.html#_run_modes), which vcpkg detects as a compiler change, resulting in binary cache invalidation and a rebuild of all dependencies. This, in turn, is problematic if you sometimes have to build without ccache, e.g. when building release binaries.

**Environment**
- OS: macOS

**To Reproduce**
1. Add `-DCMAKE_CXX_COMPILER_LAUNCHER=ccache` to cmake switches
2. Build your project in manifest mode

**Expected behavior**
vcpkg should honor the `<LANG>_COMPILER_LAUNCHER` switches and pass it down to CMake.

**Comment by Neumann-A** [comment]
> I added -DCMAKE_CXX_COMPILER_LAUNCHER=ccache to my cmake switches

vcpkg does not pass any `CMAKE_` switches. Maybe (finally) setup a custom `VCPKG_CHAINLOAD_TOOLCHAIN_FILE` ?
